library/std/src/io/readbuf.rs

```diff
@@ -243,3 +251,135 @@ impl<'a> ReadBuf<'a> {
         self.initialized
     }
 }
+
+/// A wrapper around [`&mut ReadBuf`](ReadBuf) which prevents the buffer that the `ReadBuf` points to from being replaced.
+#[derive(Debug)]
+pub struct ReadBufRef<'a, 'b> {
+    read_buf: &'a mut ReadBuf<'b>,
+}
+
+impl<'a, 'b> ReadBufRef<'a, 'b> {
+    /// Creates a new `ReadBufRef` referencing the same `ReadBuf` as this one.
+    pub fn reborrow(&mut self) -> ReadBufRef<'_, 'b> {
```

Comment:
This can't be a `Clone` impl. 
- `Clone` is `&'r ReadBufRef<'a, 'b> -> ReadBufRef<'a, 'b>`
- this is `&'r mut ReadBufRef<'a, 'b> -> ReadBufRef<'r, 'b>`